src/Engine/Renderer/Camera/Camera.cpp

```diff
@@ -132,5 +132,61 @@ Camera* Camera::duplicate( Entity* cloneEntity, const std::string& cloneName ) c
     return cam;
 }
 
+void Camera::fitZRange( const Core::Aabb& aabb ) {
+    Ra::Core::Vector3 minAabb   = aabb.min();
+    Ra::Core::Vector3 maxAabb   = aabb.max();
+    Ra::Core::Vector3 position  = m_frame.translation();
+    Ra::Core::Vector3 direction = m_frame.linear() * Ra::Core::Vector3( 0, 0, -1 );
+
+    float d;
+
+    Ra::Core::Vector3 corner = minAabb;
+    d                        = direction.dot( corner - position );
+    m_zNear                  = d;
+    m_zFar                   = d;
+
+    corner  = Ra::Core::Vector3( minAabb[0], minAabb[1], maxAabb[2] );
+    d       = direction.dot( corner - position );
+    m_zNear = std::min( d, m_zNear );
+    m_zFar  = std::max( d, m_zFar ); // beware if all corners are behind the camera
```

Comment:
Could you use a macro for this?
something like:
```c++
#define FIT_RANGE( x, y, z )                                          \
    {                                                                 \
        const auto corner = Ra::Core::Vector3( x, y, z );             \
        const auto d         =  direction.dot( corner - position );   \
        m_zNear = std::min( d, m_zNear );                             \
        m_zFar   = std::max( d, m_zFar );                             \
    }
FIT_RANGE( minAabb[0], minAabb[1], maxAabb[2] );
FIT_RANGE( minAabb[0], maxAabb[1], minAabb[2] );
FIT_RANGE( minAabb[0], maxAabb[1], maxAabb[2] );
FIT_RANGE( maxAabb[0], minAabb[1], minAabb[2] );
FIT_RANGE( maxAabb[0], minAabb[1], maxAabb[2] );
FIT_RANGE( maxAabb[0], maxAabb[1], minAabb[2] );
```

Reply:
Why not ... I'll try to find something concise and efficient.

Reply:
Done with a lambda, not a macro.